File: _verify_cloud.py
# -*- coding: utf-8 -*-
"""验证云端数据分布与新源样例。"""
import os
import json
from collections import Counter

import httpx
from dotenv import load_dotenv

load_dotenv("E:/AIMemory/DaoBao/offer-hui/apps/worker/.env")
url = os.environ["SUPABASE_URL"]
key = os.environ["SUPABASE_SERVICE_KEY"]
H = {"apikey": key, "Authorization": "Bearer " + key}


def q(path):
    r = httpx.get(url + "/rest/v1/" + path, headers=H, timeout=30)
    r.raise_for_status()
    return r.json()


rows = q("jobs?select=source")
cnt = Counter(r["source"] for r in rows)
print("总岗位:", len(rows), "| 按来源:", dict(cnt))

for s in ["fjut", "fjrclh", "fj99"]:
    sample = q(f"jobs?source=eq.{s}&select=title,company_name,city,job_type,deadline_at,apply_url,tags&limit=3&order=posted_at.desc.nullslast")
    print(f"\n--- {s} 样例 ---")
    for j in sample:
        tags = json.dumps(j.get("tags") or [], ensure_ascii=False)[:70]
        print(f"  {(j['title'] or '')[:26]} | {(j.get('company_name') or '')[:16]} | {j.get('city') or ''} | {j.get('job_type') or ''} | 截止:{(j.get('deadline_at') or '-')[:10]} | {tags}")

File: _verify_cloud.js
const fs = require("fs");
const env = fs.readFileSync("E:/AIMemory/DaoBao/offer-hui/apps/worker/.env", "utf8");
const url = env.match(/SUPABASE_URL=(.+)/)[1].trim();
const key = env.match(/SUPABASE_SERVICE_KEY=(.+)/)[1].trim();
console.log("URL:", url);
const base = url + "/rest/v1";

async function q(path) {
  const r = await fetch(base + path, {
    headers: { apikey: key, Authorization: "Bearer " + key },
  });
  if (!r.ok) throw new Error(r.status + " " + (await r.text()).slice(0, 300));
  return r.json();
}

(async () => {
  const rows = await q("jobs?select=source");
  const bySrc = {};
  rows.forEach((r) => (bySrc[r.source] = (bySrc[r.source] || 0) + 1));
  console.log("按来源:", bySrc, "总:", rows.length);
  const sample = await q("jobs?source=eq.fj99&select=title,company_name,city,job_type,deadline_at,apply_url,tags&limit=3");
  sample.forEach((j) => console.log(" ", j.title.slice(0, 26), "|", (j.company_name || "").slice(0, 16), "|", j.city, "|", j.job_type, "|", (j.deadline_at || "-").slice(0, 10), "|", JSON.stringify(j.tags).slice(0, 70)));
})();
